### PR TITLE
#716 [FE] Implement component for moderation

### PR DESCRIPTION
**Pull request by Lvyshnevska**
A moderation component has been implemented.
When the moderator presses the "Затвердити" button (currently only approve action is available) in the email, he is redirected to the main page of the site. A PATCH request with the related data is then sent.
The moderation request is processed and result is displayed in the modal window.

image

image

image

### CODE REVIEW

File: FrontEnd/src/components/moderation/ModerationModal.jsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import axios from 'axios';
+import{ useState, useEffect } from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
+import MainPage from '../landing-page/MainPage';
+import classes from './ModerationModal.module.css';
+
+
+export function ModerationModal() {
+    const { id, action} = useParams();
```
Comment (popovycholeg) [inline FrontEnd/src/components/moderation/ModerationModal.jsx:9]: nit: formatting 
```suggestion
    const { id, action } = useParams();
```

File: FrontEnd/src/components/moderation/ModerationModal.jsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import axios from 'axios';
+import{ useState, useEffect } from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
+import MainPage from '../landing-page/MainPage';
+import classes from './ModerationModal.module.css';
+
+
```
Comment (popovycholeg) [inline FrontEnd/src/components/moderation/ModerationModal.jsx:7]: nit: no need for two empty lines

File: FrontEnd/src/components/moderation/ModerationModal.jsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                    Закрити
+                    </button>
+                </div>
+            </div>
+        </div>
+      )}
+      {modalVisible && <div className={classes['blur-background']}></div>}
+    </div>
+    );
+  }
```
Comment (popovycholeg) [inline FrontEnd/src/components/moderation/ModerationModal.jsx:83]: nit: new line at the end of the file

File: FrontEnd/src/components/moderation/ModerationModal.jsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                {moderationStatus}
+            </div>
+            <div className={classes['modal__footer']}>
+                <div className={classes['button-container']}>
+                    <button
+                    className={classes['modal__button']}
+                    onClick={closeModal}
+                    >
+                    Закрити
+                    </button>
```
Comment (popovycholeg) [inline FrontEnd/src/components/moderation/ModerationModal.jsx:75]: It doesn't seem to be properly formatted. Could you format the file? I think we need to add linting github check

**Review by popovycholeg — APPROVED** [review]
With couple minor commenst

File: FrontEnd/src/components/moderation/ModerationModal.jsx (hunk 1 above)
Comment (Lvyshnevska) [reply FrontEnd/src/components/moderation/ModerationModal.jsx]: Fixed

File: FrontEnd/src/components/moderation/ModerationModal.jsx (hunk 2 above)
Comment (Lvyshnevska) [reply FrontEnd/src/components/moderation/ModerationModal.jsx]: Fixed

File: FrontEnd/src/components/moderation/ModerationModal.jsx (hunk 3 above)
Comment (Lvyshnevska) [reply FrontEnd/src/components/moderation/ModerationModal.jsx]: Fixed

File: FrontEnd/src/components/moderation/ModerationModal.jsx (hunk 4 above)
Comment (Lvyshnevska) [reply FrontEnd/src/components/moderation/ModerationModal.jsx]: Done

File: FrontEnd/src/components/moderation/ModerationModal.jsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+    const closeModal = () => {
+      setModalVisible(false);
+    };
+
+    return (
+      <div>
+      <MainPage />
+      {modalVisible && (
+        <div className={classes['modal']}>
```
[the remark this answers is not in the document]
Comment (Lvyshnevska) [reply FrontEnd/src/components/moderation/ModerationModal.jsx]: Replaced with MyModal

**Review by KuzykY — APPROVED** [review]

**Review by YanZhylavy — APPROVED** [review]

**Pull request merged**